api/go.mod

```diff
@@ -19,32 +18,27 @@ require (
 	github.com/gin-contrib/pprof v1.3.0
 	github.com/gin-contrib/static v0.0.0-20200916080430-d45d9a37d28e
 	github.com/gin-gonic/gin v1.6.3
-	github.com/gogo/protobuf v1.3.2 // indirect
-	github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e // indirect
 	github.com/google/uuid v1.1.2 // indirect
 	github.com/gorilla/websocket v1.4.2 // indirect
 	github.com/grpc-ecosystem/go-grpc-middleware v1.2.2 // indirect
-	github.com/grpc-ecosystem/grpc-gateway v1.16.0 // indirect
 	github.com/jonboulle/clockwork v0.2.2 // indirect
-	github.com/json-iterator/go v1.1.11 // indirect
 	github.com/prometheus/client_golang v1.8.0 // indirect
 	github.com/satori/go.uuid v1.2.0
 	github.com/shiningrush/droplet v0.2.6-0.20210127040147-53817015cd1b
 	github.com/shiningrush/droplet/wrapper/gin v0.2.1
 	github.com/sirupsen/logrus v1.7.0 // indirect
 	github.com/sony/sonyflake v1.0.0
 	github.com/spf13/cobra v0.0.3
-	github.com/spf13/pflag v1.0.5 // indirect
+	github.com/spf13/viper v1.8.1
```

Comment:
I remember that the license of a dependency of `viper` was not compatible with Apache 2.0,  please confirm.

Reply:
Hi, @nic-chen. I'm sorry for the slow response.

I used `google/go-licenses` to generate a list of the `spf13/viper`'s dependencies and licenses.

| library | URL | license |
|  ----  | ----  | ---- |
| github.com/mitchellh/mapstructure | https://github.com/mitchellh/mapstructure/blob/master/LICENSE | MIT |
| github.com/magiconair/properties | https://github.com/magiconair/properties/blob/master/LICENSE.md | BSD-2-Clause |
| github.com/spf13/afero | https://github.com/spf13/afero/blob/master/LICENSE.txt | Apache-2.0 |
| github.com/spf13/pflag | https://github.com/spf13/pflag/blob/master/LICENSE | BSD-3-Clause |
| github.com/hashicorp/hcl | https://github.com/hashicorp/hcl/blob/master/LICENSE | MPL-2.0 |
| github.com/pelletier/go-toml | https://github.com/pelletier/go-toml/blob/master/LICENSE | Apache-2.0 |
| github.com/spf13/viper | https://github.com/spf13/viper/blob/master/LICENSE | MIT |
| github.com/spf13/cast | https://github.com/spf13/cast/blob/master/LICENSE | MIT |
| github.com/spf13/jwalterweatherman | https://github.com/spf13/jwalterweatherman/blob/master/LICENSE | MIT |
| github.com/subosito/gotenv | https://github.com/subosito/gotenv/blob/master/LICENSE | MIT |
| github.com/fsnotify/fsnotify | https://github.com/fsnotify/fsnotify/blob/master/LICENSE | BSD-3-Clause |
| gopkg.in/yaml.v2 | https://github.com/go-yaml/yaml | Apahce-2.0 |
| golang.org/x/text | https://golang.org/x/text | BSD-3-Clause |
| gopkg.in/ini.v1 | https://gopkg.in/ini.v1 | Apache-2.0 |

And these licenses are also used in current Manager API dependencies.

Reply:
I'm not sure about `MPL-2.0`, @moonming  please have a look, thanks.

Reply:
### Description
`Viper` library uses the API of `hashicorp` open source repository using `MPL-2.0` license, but does not use its code directly.

### Solution
If importing a library using `MPL-2.0` license is not feasible, I will consider implementing a shim code and replacing it with `go mod replace`. It emulates the API of that library and provides empty results.

It's like the following image.
![image](https://user-images.githubusercontent.com/8078418/125726521-ad4d8029-a8f3-4e95-b100-917ff5895d9a.png)

So is such a solutionfeasible?

Reply:
I searched the go project under apache, and many projects also use viper
for example: 

https://github.com/apache/skywalking-banyandb

https://github.com/apache/camel-k

I have no problem here.

Reply:
MPL-2.0 maybe can NOT in apache projects: https://www.apache.org/legal/resolved.html

Reply:
@moonming Sorry to bother you again.

- I have read this specification [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html) and made a query in JIRA, I found the following discussion, [LEGAL-562: Handling dependency licenses for the Solr Operator, a go project](https://issues.apache.org/jira/browse/LEGAL-562).
I think our situation is similar to that in the article. We do not include the code under the MPL-2.0 license in the code distribution. The code is downloaded by the user during the compilation process. In the docker image involving binary distribution, we should modify the license and note files and include them in the image as a reminder. Just like these project, [apache/skywalking-banyandb](https://github.com/apache/skywalking-banyandb).

- If we really can't include the dependencies of this library, can the above scheme of replacing the dependent library code with shim? [here](https://github.com/apache/apisix-dashboard/pull/1946#discussion_r670114961)

🤔 What do you think of this? Thx.

Reply:
ping @moonming Any changes here?

Reply:
Having a dependency with a Category B license is fine